mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java

```diff
@@ -57,40 +56,8 @@ public void init() throws MessagingException {
     @Override
     public void service(Mail mail) throws MessagingException {
         MimeMessage m = mail.getMessage();
-        
         String newSubject = prefixSubject(m);
-        
-        /*
-         * Get sure to use the right encoding when add the subjectPrefix..
-         * otherwise we get problems with some special chars
-         */
-        String rawSubject = getFirstSubjectHeader(m);
-        String mimeCharset = determineMailHeaderEncodingCharset(rawSubject);
-        if (mimeCharset == null) { // most likely ASCII
-            // it uses the system charset or the value of the
-            // mail.mime.charset property if set
-            m.setSubject(newSubject, Charsets.UTF_8.displayName());
-        } else { // original charset determined
-            String javaCharset = javax.mail.internet.MimeUtility
-                    .javaCharset(mimeCharset);
-            try {
-                m.setSubject(newSubject, javaCharset);
-            } catch (MessagingException e) {
-                // known, but unsupported encoding
-                // this should be logged, the admin may setup a more i18n
-                // capable JRE, but the log API cannot be accessed from here
-                // if (charset != null) log(charset +
-                // " charset unsupported by the JRE, email subject may be
-                // damaged");
-                m.setSubject(newSubject); // recover
-            }
-            m.saveChanges();
-        }
-    }
-
-    private String getFirstSubjectHeader(MimeMessage m) throws MessagingException {
-        String delimiter = null;
-        return m.getHeader(RFC2822Headers.SUBJECT, delimiter);
+        m.setSubject(newSubject, Charsets.UTF_8.displayName());
```

Comment:
Not sure to understand. You assume you can always convert to UTF-8? And then you replace the original mail charset by UTF-8?